app/src/main/java/io/apicurio/registry/storage/impl/sql/AbstractSqlRegistryStorage.java

```diff
@@ -789,45 +790,56 @@ private Long ensureContentAndGetId(String artifactType, ContentWrapperDto conten
      */
     private void ensureContent(TypedContent content, String contentHash, String canonicalContentHash,
             List<ArtifactReferenceDto> references, String referencesSerialized) {
+
         handles.withHandleNoException(handle -> {
-            byte[] contentBytes = content.getContent().bytes();
 
             // Insert the content into the content table.
-            String sql = sqlStatements.insertContent();
             long contentId = nextContentIdRaw(handle);
 
             try {
-                handle.createUpdate(sql).bind(0, contentId).bind(1, canonicalContentHash).bind(2, contentHash)
-                        .bind(3, content.getContentType()).bind(4, contentBytes).bind(5, referencesSerialized)
+                handle.createUpdate(sqlStatements.insertContent())
+                        .bind(0, contentId)
+                        .bind(1, canonicalContentHash)
+                        .bind(2, contentHash)
+                        .bind(3, content.getContentType())
+                        .bind(4, content.getContent().bytes())
+                        .bind(5, referencesSerialized)
                         .execute();
             } catch (Exception e) {
                 if (sqlStatements.isPrimaryKeyViolation(e)) {
+                    log.debug("Content with content hash {} already exists: {}", contentHash, content);
                     return;
                 } else {
                     throw e;
                 }
             }
 
             // If we get here, then the content was inserted and we need to insert the references.
-            insertReferencesRaw(handle, contentId, references);
+            insertReferencesRaw(contentId, references);
         });
     }
 
-    private void insertReferencesRaw(Handle handle, Long contentId, List<ArtifactReferenceDto> references) {
+    private void insertReferencesRaw(Long contentId, List<ArtifactReferenceDto> references) {
```

Comment:
Can you explain this change?  Won't this move the insertion of the references into a new Transaction?  As a result, any rollback (e.g. when using dryRun=true) would fail to roll back the references?

Or am I not remembering how handles work?